core/src/chain/mod.rs

```diff
@@ -131,82 +204,109 @@ pub trait SourceChain {
     /// returns an option for the top Pair
     fn top_pair(&self) -> Option<Pair>;
     /// get the top Pair by Entry type
-    fn top_pair_type(&self, t: &str) -> Option<Pair>;
+    fn top_pair_of_type(&self, t: &str) -> Option<Pair>;
 
     /// push a new Entry on to the top of the Chain
     /// the Pair for the new Entry is automatically generated and validated against the current top
     /// Pair to ensure the chain links up correctly across the underlying table data
     /// the newly created and pushed Pair is returned in the fn Result
-    fn push_entry(&mut self, entry: &Entry) -> Result<Pair, HolochainError>;
+    fn commit_entry(&mut self, entry: &Entry) -> Result<Pair, HolochainError>;
     /// get an Entry by Entry key from the HashTable if it exists
     fn entry(&self, entry_hash: &str) -> Result<Option<Pair>, HolochainError>;
 
     /// pair-oriented version of push_entry()
-    fn push_pair(&mut self, pair: &Pair) -> Result<Pair, HolochainError>;
-    /// get a Pair by Pair/Header key from the HashTable if it exists
-    fn pair(&self, message: &str) -> Result<Option<Pair>, HolochainError>;
+    fn commit_pair(&mut self, pair: &Pair) -> Result<Pair, HolochainError>;
+
+    // /// get a Pair by Pair/Header key from the HashTable if it exists
+    fn pair(&self, pair_hast: &str) -> Result<Option<Pair>, HolochainError>;
 }
 
 impl SourceChain for Chain {
     fn top_pair(&self) -> Option<Pair> {
-        self.actor.top_pair()
+        self.chain_actor.top_pair()
     }
 
     fn set_top_pair(&self, pair: &Option<Pair>) -> Result<Option<Pair>, HolochainError> {
-        self.actor.set_top_pair(&pair)
+        self.chain_actor.set_top_pair(&pair)
     }
 
-    fn top_pair_type(&self, t: &str) -> Option<Pair> {
+    fn top_pair_of_type(&self, t: &str) -> Option<Pair> {
         self.iter().find(|p| p.header().entry_type() == t)
     }
 
-    fn push_pair(&mut self, pair: &Pair) -> Result<Pair, HolochainError> {
+    /// Whole process of authoring an entry.
+    /// 1. `validation` of the new entry using the ribosome and validation WASM code
+    /// 2. `pushing` the new entry onto the source chain, if valid
+    /// 3. `putting` the entry into the (distributed) hash table, if defined as public
+    fn commit_pair(&mut self, pair: &Pair) -> Result<Pair, HolochainError> {
```

Comment:
For a follow up PR?

Reply:
@lucksus i disagree that you should put headers in the chain

that's so out of scope here...

Reply:
The reason I asked Damien to look at the refactor ticket before doing the linking is that he would produce more code that we would need to change if we don't do the refactor first.

Reply:
I took some more time read through @lucksus's initial comment.
I like the non orchestrated version. 
I find it more understandable if we rename the actions like this:
`Commit` to `ValidateForCommit`
`VailidationResult` to `Commit`.

So the story goes:
- `invoke_commit()` issues a `ValidateForCommit(Entry)` action
- Nucleus reduces it and on success issues a `Commit(Entry)` action
- Chain/Agent reduces it and on success issues a `Put(Entry)` action
- DHT reduces it
- `invoke_commit()` listens for a succesful Put of that entry in the DHT.

The story for receiving an Entry from the network goes:
- Network issues a `ValidateForPut(Entry)` action
- Nucleus reduces it and on success issues a `Put(Entry)` action
- DHT reduces it

I believe the main misunderstanding is that for @lucksus the HashTable is the DHT.
But thats not the case in the current code architecture. Currently the HashTable is the Chain.
Meaning, it is the actual data structure holding the Chain's data. The Chain is just a way to browse the HashTable in a certain sequence.
But since we have a HashTable for the Chain, we could also use it to store the DHT.
Eric was okay with that idea. I think its better to separate and have a second HashTable for the DHT, because you would want to store/persist them differently.

Reply:
@ddd-mtl "because you would want to store/persist them differently" this is the bit i want to get deeper into because it's not obvious to me why you'd want to change the storage mechanism based on the content of the data being stored?

Reply:
@thedavidmeister Source chain data and DHT data are of different nature. So they will most likely have different functional requirements and constraints. I have Holo and also data security in mind for now. Source chain data is more critical for example. You would want to backup that in a more robust way. This will most likely translate into different technical requirements. Especially over time. So I'm thinking ahead of this a bit and thinking its best to separate them right now.

Reply:
Well yes, one thing we haven't mentioned in this discussion so far is that Holo wants to hold a users source chain on the client in the browser's local storage, but not the DHT.

Though I guess that should be possible by having different implementations of the trait, right?